app/src/main/java/com/team7/cmput301/android/theirisproject/task/AddProblemTask.java

```diff
@@ -52,16 +57,39 @@ protected String doInBackground(Problem... params) {
                     .build();
             DocumentResult res = IrisProjectApplication.getDB().execute(post);
             if (res.isSucceeded()) {
+                Bulk postBodyPhoto = new Bulk.Builder()
+                        .defaultIndex(IrisProjectApplication.INDEX)
+                        .defaultType("bodyphoto")
+                        .addAction(bulkAddPhotos(params[0].getBodyPhotos(), res.getId()))
+                        .build();
+                IrisProjectApplication.getDB().execute(postBodyPhoto);
+                Thread.sleep(500);
                 return res.getId();
             }
             else {
                 return null;
             }
         } catch (IOException e) {
             e.printStackTrace();
+        } catch (InterruptedException e) {
+            e.printStackTrace();
         }
         return null;
     }
+    /**
+     * bulkAddPhotos sets up the bulk add to the database by
+     * setting all bodyphotos as Index in Jest to execute
+     *
+     * @param photos
+     * */
+    private List<Index> bulkAddPhotos(List<BodyPhoto> photos, String problemId) {
+        ArrayList<Index> res = new ArrayList<>();
```

Comment:
Try using List<T> interface type instead of forcing it to be an ArrayList whenever possible